src/domain/dealer-price/index.types.d.ts

```diff
@@ -0,0 +1,18 @@
+type DealerPriceServiceError = import("./errors").DealerPriceServiceError
+
+interface IDealerPriceService {
+  getExchangeRateForImmediateUsdBuy(
+    amountInSatoshis: Satoshis,
+  ): Promise<UsdCents | DealerPriceServiceError>
+  getExchangeRateForImmediateUsdSell(
+    amountInUsd: UsdCents,
+  ): Promise<Satoshis | DealerPriceServiceError>
+  getExchangeRateForFutureUsdBuy(
+    amountInSatoshis: Satoshis,
+    timeToExpiryInMinutes: Minutes,
```

Comment:
I think we should have seconds timeframe instead of minutes here.

Reply:
Ok I don't mind but the at 1 minute time frame the expiry is so small that I can't price and option with non-zero premium. I will change it but it is likely too granular.

Reply:
My point was that, given the conditions this operates in, there is objectively no material differences in option prices for changes in expiry in seconds. So 0s, 15s, 60s will all have the same premium and therefore exchange rate returned. So I felt that by using seconds we imply granularity and control that does not materially exist.

Reply:
have you modeled the implied price during high volatility event (with vol at > 300% to 1000%?). I think in 99% of the time, the time value of the option will be close to zero... unless the market is moving quickly, which is precisely when the option premium should go drastically up.